cmd/bosun/conf/conf.go

```diff
@@ -1,120 +1,120 @@
-	Squelch          Squelches `json:"-"`
-	Quiet            bool
-	SkipLast         bool
-	NoSleep          bool
-	ShortURLKey      string
-	InternetProxy    string
-	MinGroupSize     int
+	GetAlerts() map[string]*Alert
+	GetAlert(string) *Alert
+	SetAlert(string, string) (string, error)
+	DeleteAlert(alertName string) error
 
-	TSDBHost             string                    // OpenTSDB relay and query destination: ny-devtsdb04:4242
-	TSDBVersion          *opentsdb.Version         // If set to 2.2 , enable passthrough of wildcards and filters, and add support for groupby
-	GraphiteHost         string                    // Graphite query host: foo.bar.baz
-	GraphiteHeaders      []string                  // extra http headers when querying graphite.
-	LogstashElasticHosts expr.LogstashElasticHosts // CSV Elastic Hosts (All part of the same cluster) that stores logstash documents, i.e http://ny-elastic01:9200. Only works with elastc pre-v2, and expects the schema to be logstash's default.
-	ElasticHosts         expr.ElasticHosts         // CSV Elastic Hosts (All part of the same cluster), i.e http://ny-elastic01:9200. Only works with elastic v2+, and unlike logstash it is designed to be able to use various elastic schemas.
-	InfluxConfig         client.Config
+	GetNotifications() map[string]*Notification
+	GetNotification(string) *Notification
 
-	AnnotateElasticHosts []string // CSV of Elastic Hosts, currently the only backend in annotate
-	AnnotateIndex        string   // name of index / table
+	GetMacro(string) *Macro
 
-	tree            *parse.Tree
-	node            parse.Node
-	unknownTemplate string
-	bodies          *htemplate.Template
-	subjects        *ttemplate.Template
-	squelch         []string
-}
+	GetLookup(string) *Lookup
 
-// TSDBContext returns an OpenTSDB context limited to
-// c.ResponseLimit. A nil context is returned if TSDBHost is not set.
-func (c *Conf) TSDBContext() opentsdb.Context {
-	if c.TSDBHost == "" {
-		return nil
-	}
-	return opentsdb.NewLimitContext(c.TSDBHost, c.ResponseLimit, *c.TSDBVersion)
+	GetSquelches() Squelches
+	AlertSquelched(*Alert) func(opentsdb.TagSet) bool
+	Squelched(*Alert, opentsdb.TagSet) bool
+	Expand(string, map[string]string, bool) string
+	GetFuncs(EnabledBackends) map[string]parse.Func
 }
 
-// GraphiteContext returns a Graphite context. A nil context is returned if
-// GraphiteHost is not set.
-func (c *Conf) GraphiteContext() graphite.Context {
-	if c.GraphiteHost == "" {
-		return nil
-	}
-	if len(c.GraphiteHeaders) > 0 {
-		headers := http.Header(make(map[string][]string))
-		for _, s := range c.GraphiteHeaders {
-			kv := strings.Split(s, ":")
-			headers.Add(kv[0], kv[1])
-		}
-		return graphite.HostHeader{
-			Host:   c.GraphiteHost,
-			Header: headers,
-		}
-	}
-	return graphite.Host(c.GraphiteHost)
+type RuleConfWriter interface {
+	BulkEdit(BulkEditRequest) error
+	GetRawText() string
+	GetHash() string
+	SaveRawText(rawConf, diff, user, message string, args ...string) error
```

Comment:
I'm ok leaving this for now, but the `rawText` abstraction may get broken if/when we add an alternate backend